pxf/pxf-api/src/main/java/org/apache/hawq/pxf/api/FilterParser.java

```diff
@@ -279,25 +282,108 @@ int safeToInt(Long value) throws FilterStringSyntaxException {
         return value.intValue();
     }
 
+    private int parseConstDataType() throws Exception {
+        if (!Character.isDigit((char) filterByteArr[index])) {
+            throw new FilterStringSyntaxException("datatype OID should follow at " + index);
+        }
+
+        String digits = parseDigits();
+
+        try {
+            return Integer.parseInt(digits);
+        } catch (NumberFormatException e) {
+            throw new FilterStringSyntaxException("invalid numeric argument at " + digits);
+        }
+    }
+
+    private int parseDataLength() throws Exception {
+        if (((char) filterByteArr[index]) != 's') {
+            throw new FilterStringSyntaxException("data length delimiter 's' expected at " +  index);
+        }
+
+        index++;
+        return parseInt();
+    }
+
+    private int parseInt() throws Exception {
+        if (index == filterByteArr.length) {
+            throw new FilterStringSyntaxException("numeric argument expected at " + index);
+        }
+
+        String digits = parseDigits();
+
+        try {
+            return Integer.parseInt(digits);
+        } catch (NumberFormatException e) {
+            throw new FilterStringSyntaxException("invalid numeric argument " + digits);
+        }
+    }
+
+    private Object convertDataType(byte[] byteData, DataType dataType) throws Exception {
+        String data = new String(byteData);
```

Comment:
creating a string out of bytes assumes a specific charset. how will that be handled ?